twitter4j-core/src/main/java/twitter4j/TwitterImpl.java

```diff
@@ -251,12 +261,46 @@ public UploadedMedia uploadMedia(String fileName, InputStream image) throws Twit
                 , new HttpParameter("media", fileName, image)).asJSONObject());
     }
 
-    @Override
-    public UploadedMedia uploadMediaChunked(String fileName, InputStream media, long size) throws TwitterException {
-		UploadedMedia uploadedMedia = uploadMediaChunkedInit(size);
-		uploadMediaChunkedAppend(fileName, media, uploadedMedia.getMediaId());
-		return uploadMediaChunkedFinalize(uploadedMedia.getMediaId());
-    }
+	@Override
+	public UploadedMedia uploadMediaChunked(String fileName, InputStream media) throws TwitterException {
+		//If the InputStream is remote, this is will download it into memory speeding up the chunked upload process 
+		byte[] dataBytes = null;
+		try {
+			dataBytes = IOUtils.toByteArray(media);
```

Comment:
Is the Apache Commons dependency really needed for this only call? I suggest replacing this with the classic InputStream to byte[] code:
```java
			ByteArrayOutputStream baos = new ByteArrayOutputStream(256 * 1024);
			byte[] buffer = new byte[32768];
			int n;
			while((n = media.read(buffer)) != -1) {
				baos.write(buffer, 0, n);
			}
```